runtime/server_http_request.go

```diff
@@ -35,33 +35,34 @@ import (
 	"github.com/julienschmidt/httprouter"
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
+	"github.com/pborman/uuid"
+	"github.com/uber-go/tally"
 	"go.uber.org/zap"
 )
 
-// ServerHTTPRequest struct manages request
+// ServerHTTPRequest struct manages server http request
 type ServerHTTPRequest struct {
 	httpRequest *http.Request
 	res         *ServerHTTPResponse
-	started     bool
 	startTime   time.Time
-	metrics     ContextMetrics
+	started     bool
 	tracer      opentracing.Tracer
 	span        opentracing.Span
 	queryValues url.Values
 	parseFailed bool
 	rawBody     []byte
 
-	ctx           context.Context
-	contextLogger ContextLogger
-	EndpointName  string
-	HandlerName   string
-	URL           *url.URL
-	Method        string
-	Params        httprouter.Params
-	Header        Header
-
-	// Deprecated: Use contextLogger instead
-	Logger *zap.Logger
+	EndpointName string
+	HandlerName  string
+	URL          *url.URL
+	Method       string
+	Params       httprouter.Params
+	Header       Header
+
+	// Logger logs entries with default fields that contains request meta info
+	Logger Logger
```

Comment:
> Can we keep the deprecated comment 

If we are moving to 0.x release, then I don't think it's necessary to have the deprecated comment.

> can we have an internal field logger rather than exported Logger?

Yes, will do